src/Numerics/ODESolvers/GenericCallbacks.jl

```diff
@@ -12,123 +12,111 @@ using ..ODESolvers
 This callback will run the function 'f()' every `time` wallclock time seconds.
 The `mpicomm` is used to syncronize runtime across MPI ranks.
 """
-struct EveryXWallTimeSeconds
+mutable struct EveryXWallTimeSeconds
     "time of the last callback"
-    lastcbtime_ns::Array{UInt64}
+    lastcbtime_ns::UInt64
     "time between callbacks"
     Δtime::Real
     "MPI communicator"
-    mpicomm
+    mpicomm::MPI.Comm
     "function to execute for callback"
-    func::Function
-    function EveryXWallTimeSeconds(func, Δtime, mpicomm)
-        lastcbtime_ns = [time_ns()]
-        new(lastcbtime_ns, Δtime, mpicomm, func)
+    func
+    "function to call to initialize the callback"
+    init
+    function EveryXWallTimeSeconds(func, Δtime, mpicomm, init = () -> nothing)
+        lastcbtime_ns = zero(UInt64)
+        new(lastcbtime_ns, Δtime, mpicomm, func, init)
     end
 end
-function (cb::EveryXWallTimeSeconds)(initialize::Bool = false)
-    # Is this an initialization call? If so, start the timers
-    if initialize
-        cb.lastcbtime_ns[1] = time_ns()
-        # If this is initialization init the callback too
-        try
-            cb.func(true)
-        catch
-        end
-        return 0
-    end
 
+function initialize!(cb::EveryXWallTimeSeconds, solver, Q, param, t0)
+    cb.lastcbtime_ns = time_ns()
+    cb.init()
```

Comment:
@kpamnany what if we changed the interface to make it so that you called `initialize!` on the callback "function"?

Of course, this means that if you wanted to use anything that required initializing, you would have to define a custom struct.

```suggestion
function initialize!(cb::EveryXWallTimeSeconds, solver, Q, param, t0)
    cb.lastcbtime_ns = time_ns()
    initialize!(cb)
```

Reply:
> Of course, this means that if you wanted to use anything that required initializing, you would have to define a custom struct.

Which would be... not easy? We can't really inherit the logic then and would have to reimplement it.

If this is not easy to do, I think we can skip it -- it's really no big deal if `init` in particular is a closure.

Reply:
okay, will leave this, can re-evaluate later.